File: code/561/solution.go
package main /*{{{*/

import (
	"fmt"
)

func Q(length int64) int64 {
	ret := int64(0)
	base := int64(904962)
	length /= 2
	for length > 0 {
		length /= 2
		ret += length * base
	}
	return ret
}

func main() {
	fmt.Println(Q(8))
	fmt.Println(Q(int64(1e12)))
}

File: code/561/solution.cpp
#include <iostream>  /*{{{*/
using namespace std; /*}}}*/
long long Q(long long a) {
    long long ret  = 0;
    long long base = 904962;
    a /= 2;
    while (a > 0) {
        a /= 2;
        ret += a * base;
    }
    return ret;
}
int main() {
    cout << Q(8) << endl;
    cout << Q(1e12) << endl;
    return 0;
}
